### ISSUE TITLE
[CoE Starter Kit - QUESTION] DLP Activation - Certain flows do not show up in DLP impact analysis, issue

### ISSUE THREAD

**Issue by accvik** [microsoft/coe-starter-kit#4488]
### What is your question?

Hi, 

We are planning to activate a DLP for the Default environment for a large client and have run the impact analysis in the DLP Editor v2 app within the CoE starter kit. We have done the same analysis for other clients earlier before activating DLPs. This is our primary way of detecting which flows and apps will break(stop working due to DLP misalignment) in that environment so we can take care of them (migrate, delete, cleanup, etc.) first.

For this client, however, we have found flows that do not show up in the "CoE Starter Kit Admin View" app under flows. I can confirm from the Power Platform Admin Center that a flow, for example, exists using the connector "Azure SQL Data Warehouse" which we plan to block. This very flow does not, however, show up in the "Power Platform Admin View" Flow inventory nor in the impact analysis from the DLP Editor v2. We believe this is due to it is a sharepoint flow created from Sharepoint. All the Sync flows are running correctly. We are on the latest CoE starter kit version (4.2 Core December)

To the question / issue:
Will the above flow stop working when we activate the DLP? If so - why does it not show up in the impact analysis? This raises possible issues for us:
Potentially, there are many sharepoint flows whose existance we are not aware of that may stop working in the Default Environment. How can we be sure we've done our homework before activating the DLP in the Default Environment if the DLP Editor V2 app impact analysis does not cover them? 

Are we missing something here or what are your suggestions of possible ways to think about this? Thanks. 

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

December 4.2

### What app or flow are you having the issue with?

DLP Editor v2

**Comment by manuelap-msft** [comment]
Hello,

1. The CoE kit syncs inventory data (e.g. number of flows, the connectors a flow uses) to Dataverse - this runs on a daily basis so it's not always real time. Some flows can take a long time to run depending on the resources in that environment.
2. The Power Platform Admin View then shows you what inventory has been synced. If you see a flow in your tenant/environment through the Power Platform Admin Center and it's not in the Power Platform Admin View, that means something has gone wrong the sync. If you're missing a flow, you can check the Admin | Sync Template v3 (Flow) for any failures. There are no limitations on collecting inventory on flows created from SharePoint - they're just manual trigger flows in the background.
3. The DLP Editor and DLP Impact Analysis rely on the data in the Dataverse table - the logic required to perform impact analysis is so complex, it can't be performed by real time fetching the resources in the tenant and it therefore relies on the previously synced data in the Dataverse tables.
4. Yes, if your flow uses a connector that you wish to block and you know this flow uses the connector, it will be impacted (suspended/stopped) even if it doesn't show in the DLP Impact Analysis. The DLP Impact Analysis is only as accurate as the data in the Power Platform Admin View that's collected via our sync flows, so if something is missing from there, we would not be able to perform impact analysis on it.

So the next action would be for us to figure out why that flow is missing from the inventory - can you find the  Admin | Sync Template v3 (Flow) run for the Default environment and check if there's any errors.

**Comment by accvik** [comment]
Hi,

Thanks for the clarification. After some troubleshooting and also in the meanwhile we updated the kit to the December version (was running on September), suddenly the missing flow appeared in the kit data and hence in the DLP editor impact analysis. Perhaps this was a sync bug just as you described that was fixed in a later version of the kit or something else may have been broken.

I learned and takeaway that:

- All flows, even if we for some reason missed them in the starter kit data sync will be affected by DLPs that are activated. 
- All type of flows should be listed in the starter kit data, otherwise something is wrong within the sync. 

Is there any other activity other than what discussed already that we should consider before applying the DLP in the default environment? Otherwise I'm happy with your input, thanks!

**Comment by accvik** [comment]
Hi again @manuelap-msft ,

Just encountered the exact same situation for another flow - I can confirm via admin.powerplatform.com that a flow exists in the default environment but I cannot see it in the CoE inventory. 

This leads me into thinking that when I assign myself to owner for the flow, something changes which causes the Sync flow (flow) to see it (since that is what happened when I posted this thread).

To answer your question previous - I only have one failed run two weeks ago, it is running successfully every day. 

![image](https://user-images.githubusercontent.com/106146275/210255246-82cea5dd-178c-4557-88a9-49e42926d061.png)

What would be the way for us going forward? I have a GUID for the flow that does not show up in the inventory.

**Comment by manuelap-msft** [comment]
Hello,

the error you are receiving is implies that you do not have access to the Environment. This error usually only happens when the account running the sync flow and performing this action does not have environment access. Can you confirm that the account running the sync flow has permanent Power Platform Admin access - granted on a permanent basis and not via PIM, as per our [prerequisites](https://learn.microsoft.com/en-us/power-platform/guidance/coe/setup#what-identity-should-i-install-the-coe-starter-kit-with)?

**Comment by accvik** [comment]
Hi,

To be clear, this error only occured once two weeks a go but you think it can be part of the problem still even though the flow is not failing daily now?

I double checked and yes the Power Platform Admin role is granted on a permanent basis, this is my account below: 
![PP role](https://user-images.githubusercontent.com/106146275/210344028-fd49ba50-f1f3-4036-978c-270f2874c644.png)

**Comment by manuelap-msft** [comment]
So what might have happened:
- the Sync flows run on a daily basis but only pick up items that are newly created or have been modified since the last sync 
- the Sync flow ran and did not have access to the environment and because of that marked flows in that environment as deleted (if the CoE kit can no longer find something that previously existed it gets marked as deleted) 
- even if you run the sync flow again with full access, if the flow is not modified / not newly created it won't mark it as un-deleted and therefore the flow (or flows) continue to be missing from the CoE kit inventory

I would recommend you run a full inventory to get back on track here:
https://learn.microsoft.com/en-us/power-platform/guidance/coe/faq#running-a-full-inventory

**Comment by accvik** [comment]
Hi,

Thanks for the input.
Ok we'll try a full inventory and let you know how it goes. Why would the account not have access to that environment though? It has been Power Platform Administrator the last 6+ months.

**Comment by accvik** [comment]
Hi again @manuelap-msft ,

After the full refresh, I can confirm the affected flow shows up under the CoE starter kit inventory and hence also when performing DLP impact analysis in the DLP editor v2. 

Would you recommend any steps going forward to assure that we track/see all the flows? Should we always perform a full sync to be completely sure about the DLP impact analysis or would you say this behavior is very rare?

Thanks.

**Comment by accvik** [comment]
Hi,

Maybe you have some input to my latest comment? @Jenefer-Monroe

Thanks.

**Comment by Jenefer-Monroe** [comment]
There are many reasons this could have happened, the sync flows are a little flawed in that they balance excessive API calls with accurate data. That's why we have all those clean up flows is to try and keep it as correct as possible with as few API calls as possible. 

If your tenant is small enough that a full inventory is not a hardship, then I would recommend doing them once a month or so, just to make sure nothing fell through the cracks. Or before you do a big push for things like a DLP governance change.

For some tenants, that takes over a week to run so for them its not an option. For them (and you :)) we are working with the product team on a more reliable system for inventory that will not require excessive API calling flows. See https://docs.microsoft.com/power-platform/admin/self-service-analytics if curious. 

I hope that helps.

**Comment by accvik** [comment]
Hi @Jenefer-Monroe,

Thank you so much. And by doing a full push this is the way to go (the link Manuela provided): https://learn.microsoft.com/en-us/power-platform/guidance/coe/faq#running-a-full-inventory ? 

Last question, what would you consider a small tenant in terms of Power Platform? This org is about 1k apps and 1k flows.

I guess we have a strategy to be totally sure then - do a full push before DLP governance changes. 

Thanks again!

**Comment by Jenefer-Monroe** [comment]
Thats correct, we call them **_full inventory runs_**, in case our terminology is useful to you. 
Yes with 1k apps and 1k flows, that would be a small enough tenant to do periodic full inventory runs. 

Glad we could help!

**Comment by Jenefer-Monroe** [comment]
closing out as no further action for starter kit team
